### DISCUSSION TITLE
Add Pinch To Zoom View For Images / Content

### DISCUSSION THREAD

**Discussion by Axemasta** [CommunityToolkit/Maui#2161] [New Feature Discussions]
An app I am building requires the ability to zoom in on an image, similar to how you would in photos app / whatsapp.

There doesn't seem to be alot of options available to Maui, I have see there is a library [Bertuzzi.MAUI.PinchZoomImage](https://github.com/TBertuzzi/Bertuzzi.MAUI.PinchZoomImage) however it hasn't been updated for 2 years and I haven't tested the user experience. There are numerous [blog posts](https://betterprogramming.pub/creating-a-zoomable-image-view-in-swift-c5ce67f17b2e) showing you how to do it for each platform, we could take inspiration from many places!

The api could be as follows:

- ImageSource
- DoubleTapEnabled (To either zoom into the tap location or reset)
- Ability to rotate?

I thought this sort of feature would be a perfect inclusion in the community toolkit

**Comment by bijington** [comment]
Sounds good to me.

**Comment by stephenquan** [comment]
Does the PinchGestureRecognizer along with the PinchToZoomContainer not work for you? See https://learn.microsoft.com/en-us/dotnet/maui/fundamentals/gestures/pinch?view=net-maui-8.0

**Comment by Axemasta** [comment]
@stephenquan This is definitely a great foundation, my idea is to ship this as a fully integrated control instead of pushing code onto the developers to implement and maintain themselves. I have also tested the UX of this tutorial and it definitely doesn't handle like the native iOS apps do in terms of how they pinch etc. The native apps will allow you to shrink the view aswell as expand it, phase 1 could even be the pinch zoom view... I just think a fully fleshed out version would be extremely beneficial to have in the toolkit as opposed to a blog!
